pkg/controller/servingcert/controller/secret_creating_controller.go

```diff
@@ -433,3 +431,24 @@ func uidsEqual(secret *corev1.Secret, service *corev1.Service) bool {
 	return secret.Annotations[api.AlphaServiceUIDAnnotation] == suid ||
 		secret.Annotations[api.ServiceUIDAnnotation] == suid
 }
+
+// Set ownership and description annotations. Return true if there are changes on
+// the two annotations
+func setSecretOwnerDescription(secret *corev1.Secret, service *corev1.Service) bool {
+	changed := false
+	if secret.Annotations == nil {
+		secret.Annotations = map[string]string{}
+		changed = true
+	}
+	// Set service CA owner annotation to generated secret
+	if len(secret.Annotations[apiannotations.OpenShiftComponent]) == 0 {
+		secret.Annotations[apiannotations.OpenShiftComponent] = api.OwningJiraComponent
+		changed = true
+	}
+	// Generate a description for generated secret if not existed already
+	if len(secret.Annotations[apiannotations.OpenShiftDescription]) == 0 {
+		secret.Annotations[apiannotations.OpenShiftDescription] = fmt.Sprintf("Secret contains a pair signed serving certificate/key that is generated by Service CA operator for service %s/%s and is annotated to the service with annotating a service resource with 'service.beta.openshift.io/serving-cert-secret-name: %s'. The certificate is valid for 2 years.", service.Name, service.Namespace, secret.Name)
```

Comment:
```suggestion
		secret.Annotations[apiannotations.OpenShiftDescription] = fmt.Sprintf("Secret contains a pair signed serving certificate/key that is generated by Service CA operator for service/%s with hostname %s.%s.svc and is annotated to the service with annotating a service resource with 'service.beta.openshift.io/serving-cert-secret-name: %s'. The certificate is valid for 2 years.", service.Name,  service.Name, service.Namespace, secret.Name)
```

willing to take as followup.